packages/cli-kit/src/public/node/api/graphql.ts

```diff
@@ -40,6 +40,24 @@ export interface CacheOptions {
   cacheStore?: LocalStorage<ConfSchema>
 }
 
+interface RefreshedTokenOnAuthorizedResponse {
+  token?: string
+}
+
+export type RefreshTokenOnAuthorizedResponse = Promise<RefreshedTokenOnAuthorizedResponse>
+
+interface SimpleUnauthorizedHandler {
+  type: 'simple'
+  handler: () => Promise<void>
+}
+
+interface TokenRefreshHandler {
+  type: 'token_refresh'
+  handler: () => RefreshTokenOnAuthorizedResponse
+}
```

Comment:
We shouldn't need a "simple" unathorized handle, it's only used from themes, be can easily modify how themes refreshes the token to use `TokenRefreshHandler`